controllers/rhmi/rhmi_controller.go

```diff
@@ -231,6 +232,33 @@ func (r *RHMIReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 		installationCfgMap = installation.Spec.NamespacePrefix + DefaultInstallationConfigMapName
 	}
 
+	installationSKU := &sku.SKU{}
+	if installation.Spec.Type == string(rhmiv1alpha1.InstallationTypeManagedApi) {
+		skuSecretName, found, err := addon.GetStringParameterByInstallType(context.TODO(), r.Client, rhmiv1alpha1.InstallationTypeManagedApi, request.NamespacedName.Namespace, "sku")
+		if err != nil {
+			return retryRequeue, errors.Wrap(err, "Error checking for SKU secret")
+		}
+
+		if !found && !installation.ObjectMeta.CreationTimestamp.Time.Before(time.Now().Add(-(1 * time.Minute))) {
+			return retryRequeue, nil
+		}
+		//!found means the param wasn't found so we want to default rather than return
+		//but don't do it until the installation object is more than a minute old in case the secret is slow to create
+		if !found && installation.ObjectMeta.CreationTimestamp.Time.Before(time.Now().Add(-(1 * time.Minute))) {
+			skuSecretName = "TWENTY_MILLION_SKU"
+		}
+		// get a configmap from the cluster
+		cm := &corev1.ConfigMap{}
+		err = r.Get(context.TODO(), types.NamespacedName{Namespace: request.Namespace, Name: "sku-config"}, cm)
```

Comment:
I'm particularly interested in feedback on this block. If no secret is found on the cluster, aka there is no sku passed via ocm param (will be the case until that functionality is added in ocm) the skuSecretName can be added. I don't like that it's hardcoded but as long as we have a sku with the same name present in the secret in the addon config this will work fine. I think ideally in the longer term, once the functionality is implemented in ocm the sku param should be required. in that case we can just return until we find a secret.

Reply:
I did a bit of refactoring in this section so I think I'm happy with it now. I will push it soon. I will leave the comment open for now as I'm still interested in the feedback.